bin/pre-commit-hook.js

```diff
@@ -3,28 +3,81 @@
 const execSync = require( 'child_process' ).execSync;
 const spawnSync = require( 'child_process' ).spawnSync;
 const chalk = require( 'chalk' );
+const fs = require( 'fs' );
+const prettier = require( 'prettier' );
+const path = require( 'path' );
+const _ = require( 'lodash' );
 
-console.log( '\nBy contributing to this project, you license the materials you contribute ' +
+console.log(
+	'\nBy contributing to this project, you license the materials you contribute ' +
 	'under the GNU General Public License v2 (or later). All materials must have ' +
-	'GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n' );
+	'GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n'
+);
 
 // Make quick pass over config files on every change
 require( '../server/config/validate-config-keys' );
 
 const files = execSync( 'git diff --cached --name-only' )
 	.toString()
 	.split( '\n' )
-	.map( ( name ) => name.trim() )
-	.filter( ( name ) => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
+	.map( name => name.trim() )
+	.filter( name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
 
+/**
+ * Returns true if the given text contains '@ format' (without the space after @).
+ * within its first docblock. False otherwise.
+ *
+ * Note: I only don't write out the actual @ f o r m a t, because then this
+ * file would also get formatted!
+ *
+ * @param {*} text text to scan for the format keyword within the first docblock
+ */
+const shouldFormat = text => {
+	const firstDocBlockStart = /\/\*\*/;
+	const firstDocBlockEnd = /\*\//;
+
+	const firstDocBlockStartIndex = _.get( firstDocBlockStart.exec( text ), 'index' );
+
+	if ( ! _.isInteger( firstDocBlockStartIndex ) ) {
+		return;
+	}
+
+	firstDocBlockEnd.lastIndex = firstDocBlockStartIndex; // docBlockEnd must come after the start
+	const firstDocBlockEndIndex = _.get( firstDocBlockEnd.exec( text ), 'index' );
+
+	if ( ! _.isInteger( firstDocBlockEndIndex ) ) {
+		return;
+	}
+
+	const firstDocBlockText = text.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 2 );
+
+	return firstDocBlockText && /@format/.test( firstDocBlockText );
```

Comment:
:question: do you find this easier/more clear than my proposal with `indexOf()`?